vermouth/rcsu/go_structure_bias.py

```diff
@@ -0,0 +1,246 @@
+        # contact map
+        for resnode in self.res_graph.nodes:
+            chain_key = self.res_graph.nodes[resnode].get('chain', None)
+            # in vermouth within a molecule all resid are unique
+            # when merging multiple chains we store the old resid
+            # the go model always references the input resid i.e.
+            # the _old_resid
+            resid_key = self.res_graph.nodes[resnode].get('_old_resid')
+            self.__chain_id_to_resnode[(chain_key, resid_key)] = resnode
+
+        return self.__chain_id_to_resnode[(chain, resid)]
+
+    def contact_selector(self, molecule):
+        """
+        Select all contacts from the contact map
+        that according to their distance and graph
+        connectivity are elegible to form a Go
+        bond and create exclusions between the
+        backbone beads of those contacts.
+
+        Parameters
+        ----------
+        molecule: :class:vermouth.Molecule
+
+        Returns
+        -------
+        list[(abc.hashable, abc.hashable, float)]
+            list of node keys and distance
+        """
+        # distance_matrix of elegible pairs as tuple(node, node, dist)
+        contact_matrix = []
+        # find all pairs of residues that are within bonded distance of
+        # self.res_dist
+        connected_pairs = dict(nx.all_pairs_shortest_path_length(self.res_graph,
+                                                            cutoff=self.res_dist))
+        for contact in self.contact_map:
+            resIDA, chainA, resIDB, chainB = contact
+            # identify the contact in the residue graph based on
+            # chain ID and resid
+            resA = self._chain_id_to_resnode(chainA, resIDA)
+            resB = self._chain_id_to_resnode(chainB, resIDB)
+            # make sure that both residues are not connected
+            # note: contacts should be symmteric so we only
+            # check against one
+            if resB not in connected_pairs[resA]:
+                # now we lookup the backbone nodes within the residue contact
+                bb_node_A = next(filter_minimal(self.res_graph.nodes[resA]['graph'], select_backbone))
+                bb_node_B = next(filter_minimal(self.res_graph.nodes[resB]['graph'], select_backbone))
+                # compute the distance between bb-beads
+                dist = np.linalg.norm(molecule.nodes[bb_node_A]['position'] -
+                                      molecule.nodes[bb_node_B]['position'])
+                # verify that the distance between BB-beads satisfies the
+                # cut-off criteria
+                if self.cutoff_long > dist > self.cutoff_short:
+                    # find the go virtual-sites for this residue
+                    # probably can be done smarter but mehhhh
+                    atype_a = next(get_go_type_from_attributes(self.res_graph.nodes[resA]['graph'],
+                                                               resid=resIDA,
```

Comment:
the kwarg should be `_old_resid` in case of multiple chains with different chain IDs